src/main/java/it/eng/idsa/businesslogic/processor/sender/registration/SenderCreateDeleteMessageProcessor.java

```diff
@@ -1,19 +1,26 @@
 package it.eng.idsa.businesslogic.processor.sender.registration;
 
 import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.ApplicationEventPublisher;
 import org.springframework.stereotype.Component;
 
 import de.fraunhofer.iais.eis.Message;
+import it.eng.idsa.businesslogic.audit.TrueConnectorEvent;
+import it.eng.idsa.businesslogic.audit.TrueConnectorEventType;
 import it.eng.idsa.businesslogic.service.SelfDescriptionService;
 
 @Component
 public class SenderCreateDeleteMessageProcessor extends AbstractCreateRegistrationMessage {
 
 	@Autowired
 	private SelfDescriptionService selfDescriptionService;
+	
+	@Autowired
+	private ApplicationEventPublisher publisher;
 
 	@Override
 	public Message getConnectorMessage() {
+		publisher.publishEvent(new TrueConnectorEvent(TrueConnectorEventType.CONNECTOR_BROKER_DELETE, null));
```

Comment:
Can we extend this abstract method with multipartMessage so we can pass it to publisher? To avoid null pass
If applicable, also make change for other Broker processors

Reply:
Created new method to handle publishing audit events, including correlationId